src/test/java/com/synopsys/integration/jenkins/detect/substeps/ParseDetectArgumentsTest.java

```diff
@@ -0,0 +1,188 @@
+package com.synopsys.integration.jenkins.detect.substeps;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import com.synopsys.integration.jenkins.JenkinsVersionHelper;
+import com.synopsys.integration.jenkins.detect.DetectJenkinsEnvironmentVariable;
+import com.synopsys.integration.jenkins.extensions.JenkinsIntLogger;
+import com.synopsys.integration.log.LogLevel;
+import com.synopsys.integration.util.IntEnvironmentVariables;
+
+import hudson.model.TaskListener;
+
+public class ParseDetectArgumentsTest {
+
+    private final static IntEnvironmentVariables intEnvironmentVariables = Mockito.mock(IntEnvironmentVariables.class);
+    private final static JenkinsVersionHelper jenkinsVersionHelper = Mockito.mock(JenkinsVersionHelper.class);
+
+    private final static String errorMessage = "Output Detect Command does not contain: ";
+    private final static String loggingLevelKey = "--logging.level.com.synopsys.integration";
+    private final static String expectedJavaHome = "/usr/bin/java";
+    private final static String expectedDetectPath = "/usr/bin/detect.jar";
+    private final static String expectedJenkinsVersion = "JenkinsVersion";
+    private final static String expectedJenkinsPluginVersion = "JenkinsPluginVersion";
+    private final static String expectedCustomLogLevel = "TestLogLevel";
+
+    private final static String jenkinsVersionParam = "--detect.phone.home.passthrough.jenkins.version";
+    private final static String pluginVersionParam = "--detect.phone.home.passthrough.jenkins.plugin.version";
+
+    private final TaskListener taskListener = Mockito.mock(TaskListener.class);
+    private final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    private final JenkinsIntLogger jenkinsIntLogger = new JenkinsIntLogger(taskListener);
+
+    private Map<String, String> expectedVisibleDetectProperties = new LinkedHashMap<>();
+    private Map<String, String> expectedHiddenDetectProperties = new LinkedHashMap<>();
+
+    @BeforeEach
+    public void setUp() {
+        jenkinsIntLogger.setLogLevel(LogLevel.DEBUG);
+
+        // Set default expected values. Tests will apply different if needed
+        Mockito.when(taskListener.getLogger()).thenReturn(new PrintStream(byteArrayOutputStream));
+        Mockito.when(intEnvironmentVariables.getValue(DetectJenkinsEnvironmentVariable.SHOULD_ESCAPE.stringValue(), "true")).thenReturn("true");
+        Mockito.when(jenkinsVersionHelper.getJenkinsVersion()).thenReturn(java.util.Optional.of(expectedJenkinsVersion));
```

Comment:
Why aren't we importing Optional here?

Reply:
Grr... I am unsure what I am doing that causes IntelliJ to do this VS importing. I missed this in review, I will fix.